public/app/shared/utilService.js

```diff
@@ -0,0 +1,172 @@
+     * as a BasicLTI tool, it will scroll the parent window to the specified position
+     *
+     * @param  {Number}           position            The vertical scroll position to scroll to
+     */
+    var scrollTo = function(position) {
+      // When running Collabosphere as a BasicLTI tool, scroll the parent window
+      if (window.parent) {
+        postIFrameMessage(function() {
+          return {
+            subject: 'changeParent',
+            scrollTo: position
+          };
+        });
+      // Otherwise, scroll the current window
+      } else {
+        window.scrollTo(0, position);
+      }
+    };
+
+    /**
+     * Get the current scroll position. When Collabosphere is being run stand-alone,
+     * it will return the scroll position of the current window. When Collabosphere is being run
+     * as a BasicLTI tool, it will return the scroll position of the parent window
+     *
+     * @return {Promise<Number>}                      Promise returning the current scroll position
+     */
+    var getScrollPosition = function() {
+      var deferred = $q.defer();
+      // When running Collabosphere as a BasicLTI tool, request the scroll position of the parent window
+      if (window.parent) {
+        postIFrameMessage(function() {
+          return {
+            subject: 'getScrollPosition'
+          };
+        });
+
+        // The parent window will respond with a message into the current window containing
+        // the scroll position of the parent window
+        window.onmessage = function(ev) {
+          if (ev && ev.data) {
+            var message;
+            try {
+              message = JSON.parse(ev.data);
+            } catch (err) {
+              // The message is not for us; ignore it
+              return;
+            }
+            if (message.scrollPosition !== undefined) {
+              deferred.resolve(message.scrollPosition);
+            }
+          }
+        };
+      // Otherwise, retrieve the scroll position of the current window
+      } else {
+        var scrollPosition = (window.pageYOffset || document.documentElement.scrollTop) - (document.documentElement.clientTop || 0);
+        deferred.resolve(scrollPosition);
+      }
+      return deferred.promise;
+    };
+
+    /**
+     * Utility function used to send a window event to the parent container. When running
+     * Collabosphere as a BasicLTI tool, this is our main way of communicating with the container
+     * application
+     *
+     * @param  {Function}   messageGenerator              Function that will return the message to send to the parent container
+     * @api private
+     */
+    var postIFrameMessage = function(messageGenerator) {
+      // Only try to send the event when a parent container is present
+      if (window.parent) {
+        // Wait until Angular has finished rendering items on the screen
+        setTimeout(function() {
```

Comment:
@nicolaasmatthijs could you try whether `$scope.$evalAsync()` works here? http://www.bennadel.com/blog/2605-scope-evalasync-vs-timeout-in-angularjs.htm